xapian-core/cluster/Makefile.mk

```diff
@@ -1,3 +1,4 @@
 lib_src +=\
 	cluster/cluster.cc \
-	cluster/cosine_sim.cc
\ No newline at end of file
+	cluster/cosine_sim.cc \
+	cluster/round_robin.cc
```

Comment:
The missing blank line at the end of the file is causing the diff to be more complex. It's not caught in our check-patch tool, which is what's causing your CI failure. You should fix both (use fixup commits, then autosquash them back together again and force push: I'm not going to review until the PR is green).